src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp

```diff
@@ -153,11 +153,19 @@ namespace {
             for (const auto& name : wellnames) {
                 auto well2 = std::shared_ptr<Well>(new Well( this->getWell(name, handlerContext.currentStep)));
                 auto connections = std::shared_ptr<WellConnections>( new WellConnections( well2->getConnections()));
-                connections->loadCOMPDAT(record, handlerContext.grid, handlerContext.fieldPropsManager);
+                connections->loadCOMPDAT(record, handlerContext.grid, handlerContext.fieldPropsManager, handlerContext.keyword.location());
                 if (well2->updateConnections(connections, handlerContext.grid, handlerContext.fieldPropsManager.get_int("PVTNUM"))) {
                     this->updateWell(std::move(well2), handlerContext.currentStep);
                     wells.insert( name );
                 }
+
+                if (connections->size() == 0 && well2->getConnections().size() == 0) {
```

Comment:
Any reason not to use `WellConnections::empty()` here?

Reply:
No reason - I'll update